mergekit/plan.py

```diff
@@ -85,16 +84,36 @@ def __init__(
         if config.base_model and config.align_weights:
             self._space_planner = SpacePlanner(config.base_model)
 
-    @lru_cache
-    def model_arch_info(self, model: ModelReference):
-        return ConfiguredArchitectureInfo(
-            info=self.arch_info,
-            config=model.config(trust_remote_code=self.trust_remote_code),
-        )
+        self.arch_dict: Dict[ModelReference, ConfiguredArchitectureInfo] = {}
```

Comment:
Slight style nitpick: I tend to prefer having type annotations for members in the top level of the class (like in lines 47-50) instead of at first assignment like you have here.